src/subcommand/init.rs

```diff
@@ -119,12 +130,12 @@ z() {
     if [ "$#" -eq 0 ]; then
         _z_cd ~ || return "$?"
     elif [ "$#" -eq 1 ] && [ "$1" = '-' ]; then
-        _z_cd ~- || return "$?"
```

Comment:
Removing this will cause the new path to be echoed in some shells. This essentially means that for those who have enabled `_Z_ECHO`, the new path will be echoed twice.

Reply:
This is either a Bashism, or a bug in `mrsh` (I should get around to filing these sometime soon). I can't find any documentation about `cd ~-` other than in the context of `bash`, so (for now and until this is made clear) wouldn't `cd "$@" >/dev/null` above do the same thing? Errors would still print, so I personally don't see a problem with it.

Reply:
Yeah, `cd ~-` isn't POSIX.

Reply:
`~-` is substituted by bash with `$OLDPWD` it seems.

Reply:
@emersion ideally, we should be able to replace this with `cd ${OLDPWD}` with no problems, correct?

Reply:
Yeah, I think these are equivalent.

Reply:
Except for the case where `$OLDPWD` is null because it hasn't been set yet -- `cd -` in `bash` and `mrsh` report an error (`cd: OLDPWD is not yet` and similar), while `dash` appears to just `cd .` (thus setting `$OLDPWD`).

Reply:
Seems like this isn't correct with `~-` either:

```
> bash
$ echo ~-
~-
$ cd ~-
```

Reply:
I've opted to branch based upon the value `$OLDPWD` (or lack thereof), allowing us to report an error in the case that it is unset (like `mrsh` and `bash --posix` do).